### DISCUSSION TITLE
How do I write a function to toggle between one and two masters?

### DISCUSSION THREAD

**Discussion by port19x** [awesomeWM/awesome#3831] [Q&A] [answered]
I tried to write a function myself, but didn't get anywhere.

Here is where I left off
```lua
function toggletwo()
    t = ascreen.focused().selected_tag
    c = tag.getproperty(t, "master_count")
    if c >= 2 then
       x = -1
    else
       x = 1
    end
    awful.tag.incnmaster(x)
end
```

My goal is to bind this to a key so that I can effectively toggle between a "grid layout" and a regular master & stack

**Comment by Aire-One** [comment, marked as answer]
Hello @port19x 

If you want to want a specific number of master clients for given tag, the easiest way to do it is to directly assign to the tag `master_count` https://awesomewm.org/apidoc/core_components/tag.html#master_count

```lua
function toggletwo()
    t = ascreen.focused().selected_tag
    t.master_count = t.master_count == 1 and 2 or 1
end
```

> My goal is to bind this to a key so that I can effectively toggle between a "grid layout" and a regular master & stack

You probably should be looking for a grid layout instead 🤔 It shouldn't be that hard to write a custom layout.

https://github.com/xinhaoyuan/layout-machi can be used to generate on the fly layouts.
